### DISCUSSION TITLE
How to hide tray icons?

### DISCUSSION THREAD

**Discussion by wsw70** [valinet/ExplorerPatcher#1504] [Q&A] [answered]
I have just installed ExplorerPatcher (22621.1555.55.1) on my Windows 11 (22H2, build 22621.1702) and when the installation was done two visible things happened:

- the taskbar icons got separated, no more grouping → THANK YOU, this is what I was looking for
- the tray icons are now all visible, taking half of the lower bar space.  I do not know how to hide some of them.

Is there a setting to enable the ability to hide (some) systray icons?

**Comment by pyrates999** [comment, marked as answer]
If you haven't switched over to the windows 10 start menu:

1. right click on the taskbar and select taskbar settings.
2. Then under notification area, click on **select which icons appear on the taskbar**.
3. There's also **turn system icons on and off**.

If you did change to the windows 10 start menu, these settings don't work in windows 11 unfortunately due to Microsoft not intending you to ever use the windows 10 start menu in windows 11.  All you can do is to show all icons in the system tray or let the system automatically hide ones that it thinks you don't use.

I prefer to show all icons in the system tray.

**Reply by wsw70** [reply]
Thank you for the reply. 

If I understand you correctly, the 
- "_If you haven't switched over to the Windows 10 start menu_" means that I stay with the Windows 11 taskbar (which is equivalent to not having ExplorerPatcher, right? Or maybe ExplorerPatcher adds some features/settings outside of what is available from within Windows 11, but not related to grouping)
- "_If you did change to the windows 10 start menu_" means that I did that in ExplorerPatcher, but in that case I cannot control the taskbar area anymore.

Which ultimately means that I can 
- **either** go with ungrouped apps + all the tray icons
- **or**  control the tray icons but have grouped apps (the Windows 11 way)

but never both.

**Comment by pyrates999** [comment]
StartAllBack has the option to allow for ungrouping of apps in the taskbar according to this:

https://www.groovypost.com/howto/ungroup-taskbar-in-windows-11/

So perhaps use explorer patcher for other features that it has.

**Comment by AccroG33K** [comment]
When you have the windows 10 layout and all tray icons are hidden under the little arrow button, you can then drag tray icons and drop them on the taskbar, allowing you to put them visible at all times.
This has been useful for me because stupid laptop doesn't have any indicators for Caps lock, num lock etc, so i have to use traystatus as a workaround. Not showing it on the taskbar was a real dealbreaker for me.

**Reply by bdurrer** [reply]
Thanks, this is the solution for having specific icons!

Meanwhile, the ExplorerPatcher Settings offers an option "Customize notification area icons" that leads to an oldschool-styled Window which resets as soon as you hit save:

![grafik](https://github.com/valinet/ExplorerPatcher/assets/508289/d5dd18af-5cb5-4d6b-8956-7e8437596a6b)

**Comment by ninja-gaijin** [comment]
I have the same issue.
I want the 'windows 10 taskbar style' 
However, I also want my system tray icons to hide the ones that I don't care about seeing all the time. With the little arrow to show them.
Perhaps this could be added to the ExplorerPatcher options?

**Reply by pyrates999** [reply]
That's controlled by Microsoft and they have not put in the code to do what you want.

**Reply by ninja-gaijin** [reply]
As Vipin2003 said below, it is possible to fix:
You can simply set tray icons to hide which will not take ur taskbar space. Simply follow these steps :

R-click on clock and select "Customize notification icons"
In new window, you will see a new option "Always show all icons and notification". Simply uncheck this.
This will completely solve ur problem. I know that i'm late but I see this today.

----

Just an update - I'm not sure why the advice worked for me the other day, but since I have reinstalled Windows 11 and am redoing the process, I noticed that it is enabled via a feature in ExplorerPatcher itself - in taskbar settings of explorerpatcher, the third option down is 'customise notification area icons' and this is where I click 'always show all icons and notifications on the taskbar' to make it work.

**Comment by vippium** [comment]
> I have just installed ExplorerPatcher (22621.1555.55.1) on my Windows 11 (22H2, build 22621.1702) and when the installation was done two visible things happened:
> 
> * the taskbar icons got separated, no more grouping → THANK YOU, this is what I was looking for
> * the tray icons are now all visible, taking half of the lower bar space.  I do not know how to hide some of them.
> 
> Is there a setting to enable the ability to hide (some) systray icons?

You can simply set tray icons to hide which will not take ur taskbar space. Simply follow these steps : 
1. R-click on clock and select "Customize notification icons"
2. In new window, you will see a new option "Always show all icons and notification". Simply uncheck this.

This will completely solve ur problem. I know that i'm late but I see this today.

**Reply by ninja-gaijin** [reply]
Thanks Vipin2003, that did exactly what I want!!! Love hiding some icons. Legend!!

**Reply by vippium** [reply]
It's okay bro.. I got ur issur today. Sorry for late reply. I try to answer asap

**Reply by ninja-gaijin** [reply]
Really appreciate it! Obviously I would have survived with the long bar of icons but it's so good to not have the clutter :)

**Reply by vippium** [reply]
Close the issue if resolved...Thanks

**Reply by ninja-gaijin** [reply]
I'm not the original creator, who marked another post as the answer

**Reply by pyrates999** [reply]
Did Microsoft fix this so you can use the windows 10 taskbar and select which tray icons you want hidden?

**Reply by vippium** [reply]
That setting in "Taskbar settings" are only have to work with win11 taskbar. You have to use EP to customize win10 taskbar. Still u can set which icon you want to show in taskbar..

**Reply by ninja-gaijin** [reply]
Just an update - I'm not sure why the advice worked for me the other day, but since I have reinstalled Windows 11 and am redoing the process, I noticed that it is enabled via a feature in ExplorerPatcher itself - in taskbar settings of explorerpatcher, the third option down is 'customise notification area icons' and this is where I click 'always show all icons and notifications on the taskbar' to make it work.

**Reply by vippium** [reply]
Both are same...

**Comment by gggirlgeek** [comment]
It works on latest Win11Pro 23H2 22631.2506 (Developer Mode enabled, Test Mode disabled, Driver Signing enforced, Integrity Checks enabled -- In other words: just default Developer Mode.)  My Setting for Taskbar Behavior >> "...Combine" is:  "When Tasbar is full". ("Never" works fine too.)

ExplorerPatcher v. 22621.2506.60.1

Note: I don't think Developer Mode is needed. I just enabled it for something else I use (symlinks.)  


### My Steps:
**I installed ExplorerPatcher _As Admin_**, to normal C:\Program Files. I opened its settings and put the taskbar in Windows 10 style. **I restarted my entire computer**, just for good measure. I opened ExplorerPatcher settings again (Taskbar right-click, Properties) I changed all other settings I wanted. **Then I restarted Explorer.exe. I restarted the computer again too, JIK_**. THEN, from inside ExplorerPatcher's settings, **_I opened "Customize Notification Area Icons". I UNCHECKED "Always Show Tray Icons" checkbox**. This gave me the hidden System Tray and its arrow to move tray-icons as I please. See screenshots below.

<img width="490" alt="Hiding tray icons - Explorer Patcher1" src="https://github.com/valinet/ExplorerPatcher/assets/2652614/25ef6ec5-a148-41e3-9ad6-c7f823f2e8d8">
<img width="631" alt="Hiding tray icons - Explorer Patcher2" src="https://github.com/valinet/ExplorerPatcher/assets/2652614/b85c3196-66fb-49cf-a583-7f256f78eade">
<img width="312" alt="Taskbar on the side of the screen ExplorerPatcher" src="https://github.com/valinet/ExplorerPatcher/assets/2652614/5e0b88f9-2a92-4f59-92fd-c46905fc7e30">  


### Extra measures that might help if the above steps don't work:  

*Run ExplorerPatcher's settings Elevated: Create a shortcut to ExplorerPatcher's settings, anywhere, and set it to Run As Admin (Right-click >> Properties >> Advanced >> Check Run as Administrator.) Double-click to run it.
_This is the command to put in the shortcut:_
`C:\Windows\System32\rundll32.exe "C:\Program Files\ExplorerPatcher\ep_gui.dll",ZZGUI`  

*Make sure your User Account is an Administrator type account, not Standard User -- Meaning you are part of the "Administrators Group. If not, you have to log into an Administrator Type account and change your current Account Type temporarily and reinstall ExplorerPatcher, and set it up how you want it.  

_*Last resort:_ When I installed and set up ExplorerPatcher, I had the same Privileges as the account called "Administrator" (different from Administrator Account Type, or Elevating "As Admin".) To get these privileges temporarily, change 2 Local/Group Policies, reboot, make your ExplorerPatcher changes including rebooting as many times as needed. Change the Local/Group Policies back again (because it is vulnerable to malware and hackers if you don't.) Reboot.

### Here's how:
-Start >> type "gpedit" >> hit enter when Group or Local Policy editor is shown -- either if fine.
-Go to: Computer Configuration > Windows Settings > Security Settings > Local Policies > Security Options
Open: **"User Account Control: Turn on Admin Approval Mode"**
Set it to: **Disabled**
Close the Group Policy editor window.
Open a command prompt As Admin (Start >> Type "CMD" >> right-click >> select Run as Administrator)
Type: `gpupdate /force`
Close the CMD window.
Reboot.
Reinstall ExplorerPatcher and go through the setup/reboot process described above again.
Once satisified:
**CHANGE THE GROUP POLICY SETTING BACK!!... or the computer will bite you!!!**


Thanks so much, Devs! I'm so happy to be using Window 11 in peace, at last!

**Comment by wsw70** [comment]
> My Setting for Taskbar Behavior >> "...Combine" is: "When Tasbar is full". ("Never" works fine too.

Yes, I found that MS finally added back the feature with 23H2, no other settings needed
